gateway/api_definition.go

```diff
@@ -270,6 +270,13 @@ func (s *APISpec) Release() {
 	if s.JSVM.VM != nil {
 		s.JSVM.DeInit()
 	}
+
+	if s.HTTPTransport != nil {
+		// Prevent new idle connections to be generated.
+		s.HTTPTransport.transport.DisableKeepAlives = true
+		s.HTTPTransport.transport.CloseIdleConnections()
+		s.HTTPTransport = nil
```

Comment:
`curSpec.Release()` is called in a loop during API reload.
it's looping through old spec and releasing.
the specs are not replaced yet, and during the switchover if a request comes in, it would try to access the http transport, and this might end up in a nil pointer dereference? 
cc: @titpetric @buger

Reply:
`if s.HTTPTransport != nil {` is inside spec.Release(). 
what I'm concerned is about where HTTPTransport is actually being used in `(p *ReverseProxy) WrappedServeHTTP` 
https://github.com/TykTechnologies/tyk/blob/master/gateway/reverse_proxy.go#L1243
looking at https://github.com/TykTechnologies/tyk/blob/master/gateway/reverse_proxy.go#L1353 we can see it won't end up in nil pointer dereference, but might end up creating a new transport.